packages/wdio-browser-runner/src/index.ts

```diff
@@ -46,20 +46,20 @@ export default class BrowserRunner extends LocalRunner {
         await super.initialise()
     }
 
-    run (runArgs: RunArgs) {
+    async run (runArgs: RunArgs) {
         runArgs.caps = makeHeadless(this.options, runArgs.caps)
 
         if (runArgs.command === 'run') {
             runArgs.args.baseUrl = this._config.baseUrl
         }
 
-        const worker = super.run(runArgs)
-        this.#server.on('debugState', (state: boolean) => worker.postMessage('switchDebugState', state))
-        this.#server.on('workerHookExecution', (payload: HookTriggerEvent) => {
+        const worker = await super.run(runArgs)
+        this.#server.on('debugState', async (state: boolean) => await worker.postMessage('switchDebugState', state))
+        this.#server.on('workerHookExecution', async (payload: HookTriggerEvent) => {
```

Comment:
Making these `async` functions makes no difference as they are called in an event handler